libsql/src/database/builder.rs

```diff
@@ -342,6 +357,48 @@ cfg_replication! {
                 crate::util::ConnectorService::new(svc)
             };
 
+            #[cfg(feature = "sync")]
+            {
+                use super::SyncProtocol;
+                match sync_protocol {
+                    p @ (SyncProtocol::Auto | SyncProtocol::V2) => {
+                        let client = hyper::client::Client::builder()
+                            .build::<_, hyper::Body>(connector.clone());
+
+                        let req = http::Request::get(format!("{url}/sync/0/0/0"))
+                            .header("Authorization", format!("Bearer {}", auth_token))
+                            .body(hyper::Body::empty())
+                            .unwrap();
+
+                        let res = client
+                            .request(req)
+                            .await
+                            .map_err(|err| crate::Error::Sync(err.into()))?;
```

Comment:
I am merging this, but I think if client explicitly selects V2, we should skip the probing. We anyway need this for offline use case where you _cannot_ probe over the network at database open.